examples/nlp/language_modeling/conf/megatron_gpt_prompt_learning_config.yaml

```diff
@@ -104,6 +105,6 @@ model:
       name: CosineAnnealing
       warmup_steps: 50
       constant_steps: 10
-      min_lr: 1e-6
+      min_lr: 0.0 # min_lr must be 0.0 if pipeline_parallel > 1
```

Comment:
why is this?

Reply:
To keep the gpt model frozen and use DDP with pipeline parallel models, I had to make two optimizer param groups. In the frozen gpt model optimizer param groups, I force the learning rate to be 0.0 all through training so the weights that should be frozen don't get updated. If the min_lr is > 0.0, error will get raised saying min_lr > initial lr.